src/Agent.Worker/TaskCommandExtension.cs

```diff
@@ -232,6 +236,33 @@ private void ProcessTaskUploadSummaryCommand(IExecutionContext context, string d
             }
         }
 
+        private void ProcessTaskUploadFileCommand(IExecutionContext context, string data)
+        {
+            if (!String.IsNullOrEmpty(data))
+            {
+                var filePath = Path.GetFullPath(data);
+
+                if (File.Exists(filePath))
+                {
+                    var fileName = Path.GetFileName(filePath);
+                    var properties = new Dictionary<string, string>();
+
+                    properties.Add(TaskAddAttachmentEventProperties.Type, CoreAttachmentType.FileAttachment);
+                    properties.Add(TaskAddAttachmentEventProperties.Name, fileName);
+
+                    ProcessTaskAddAttachmentCommand(context, properties, filePath);
+                }
+                else
+                {
+                    throw new Exception(string.Format(CultureInfo.InvariantCulture, "Unable to find file to upload from: {0}.", filePath));
+                }
+            }
+            else
+            {
+                throw new Exception("Cannot upload file because file location is not specified.");
```

Comment:
None of the strings in the file is localized. I moved all of them to the strings.json.